sysroot/usr/include/kernel/dev/serial/rs232.hpp

```diff
@@ -44,4 +45,14 @@ void px_rs232_init(uint16_t com_id);
  * serial output.
  * @param str Input string to be printed.
  */
-void px_rs232_print(char* str);
\ No newline at end of file
+void px_rs232_print(char* str);
+
+/**
+ * @brief Initializes the serial input buffer and
+ * returns a pointer to said buffer if successful.
+ * 
+ * @param size Buffer capacity (in bytes)
+ * @return px_ring_buff_t* Pointer to buffer. NULL
+ * if memory could not be allocated.
+ */
+px_ring_buff_t* px_rs232_init_buffer(int size);
```

Comment:
I feel weird about the serial subsystem returning a reference to it's internal data structure. Perhaps we should write an abstraction layer over the circular buffer?

Reply:
If you would like to do that then you can, but it would basically be creating duplicate functions that already exist in the ring buffer library, so I figured it would be easier to just use those with the returned pointer.

Reply:
Right, but this way we can control how the user interacts with the serial library. I know it feels like duplicated code/functions, but such is the pain of good abstractions.

Reply:
What functions would you like to be added then?

Reply:
I added get char, get string, and close pipe functions.